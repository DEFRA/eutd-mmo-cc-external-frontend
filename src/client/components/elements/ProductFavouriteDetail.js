import React from 'react';
import LinkTargetBlank from '../../components/elements/LinkTargetBlank';
import { withTranslation } from 'react-i18next';


export class ProductFavouriteDetail extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      isOpen: false
    };
  }

  onClick() {
    const toggle = !this.state.isOpen;
    this.setState({ isOpen: toggle });
  }

  render() {
    const { t } = this.props;
    return (
      <details>
        <summary onClick={() => this.onClick()} className="govuk-details__summary">{t('ccProductFavouriteDetailSummaryText')}</summary>
        <div className={`govuk-details__text ${this.state.isOpen ? 'open' : ''}`}>
          <p> 
            {t('ccProductFavouriteDetailSummaryParagraphText')}
          </p>
          <p>
            {t('ccProductFavouriteDetailSummary2ParagraphText')} <LinkTargetBlank href="/manage-favourites" ariaLabel="Opens link for information on fish export service" text={t('ccProductFavouriteDetailSummary3ParagraphText')} /> {t('ccProductFavouriteDetailSummary4ParagraphText')}
          </p>
        </div>
      </details>
    );
  }
}

export default withTranslation() (ProductFavouriteDetail);