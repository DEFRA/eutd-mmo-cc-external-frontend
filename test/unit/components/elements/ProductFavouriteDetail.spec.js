import { mount } from 'enzyme';
import * as React from 'react';
import ProductFavouriteDetail from '../../../../src/client/components/elements/ProductFavouriteDetail';
import { render } from '@testing-library/react';

describe('ProductFavouriteDetail', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = mount(<ProductFavouriteDetail />);
  });

  it('should render description', () => {
    expect(wrapper).toBeDefined();
  });


  it('should generate snapshot for the Plane Details page', () => {
    const { container } = render(wrapper)
    
    expect(container).toMatchSnapshot();
  })

  it('should handle click event', () => {
    expect(wrapper.find('ProductFavouriteDetail').state('isOpen')).toEqual(false);
    wrapper.find('ProductFavouriteDetail summary').simulate('click');
    expect(wrapper.find('ProductFavouriteDetail').state('isOpen')).toEqual(true);
  });

  it('should render page with the correct links and text', () => {
    expect(wrapper.find('a[href="/manage-favourites"]').text()).toEqual('manage your product favourites(opens in new tab)');
  });

});
