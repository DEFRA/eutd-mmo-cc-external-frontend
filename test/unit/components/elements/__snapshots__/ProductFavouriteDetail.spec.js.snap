// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductFavouriteDetail should generate snapshot for the Plane Details page 1`] = `
<div>
  <details>
    <summary
      class="govuk-details__summary"
    >
      What are product favourites?
    </summary>
    <div
      class="govuk-details__text "
    >
      <p>
        Product favourites are essential to enable the uploading of products and can also be used to speed up the process of adding products manually.
      </p>
      <p>
        You can
         
        <a
          aria-label="Opens link for information on fish export service"
          href="/manage-favourites"
          rel="noopener noreferrer"
          target="_blank"
        >
          manage your product favourites
          <span
            class="govuk-visually-hidden"
          >
            (opens in new tab)
          </span>
        </a>
         
        at any time through the favourites link in the main navigation at the top of the page.
      </p>
    </div>
  </details>
</div>
`;
